### PR TITLE
Improved browser language (navigator.language) support

### LINKED ISSUE #3882: Error: Couldn't find translations for ru-RU
Hi everyone!

Screenshot at Apr 18 17-58-48

### PR DESCRIPTION
**Pull request by iquidus**
589b174
On a users first visit (no cookie present) the language cookie was being set to padOptions.lang (from settings.json on server). Removes this behavior, allowing l10n to localize based on priority (cookie, navigator.language, navigator.userLanguage, 'en'), the users first visit will instead have no language cookie, giving browser language priority unless another language is selected in the menu (setting language cookie)

77a3059
Better handling of locale variants (e.g when browser language is ru-RU)
closes #3882

### CODE REVIEW

**Comment by JohnMcLear** [comment]
Needs a rebase :)

**Comment by iquidus** [comment]
Travis CI failing with: SAUCE_USERNAME is unset - exiting. (env vars need to be set)

**Comment by JohnMcLear** [comment]
Yea, we're aware of that, don't worry about it.  It's a quirk with how we do automated testing.

**Comment by JohnMcLear** [comment]
What happens here if the user changes their language and that is stored in the cookie?  Is it respected?

**Comment by iquidus** [comment]
yes, priority has not changed. Cookie language still takes priority over browser language, it's just not set to the server lang default when theres no cookie present, allowing browser language to apply.

**Comment by iquidus** [comment]
This was already the case for the home page. However pads were getting overridden to server default as if there was no existing language cookie, one was being set, using lang from settings.json.

**Comment by JohnMcLear** [comment]
+1 should be good to merge @iquidus did you run the front end tests btw?  ''/tests/frontend'' in browser.  Just wanna be sure we don't miss something obvious :D

**Comment by iquidus** [comment]
passes: 96
failures: 0 
duration: 255.47s 
100%

**Comment by muxator** [comment]
This seems sufficiently small to pose no risk if integrated.

Scheduling for **1.8.3**.

File: src/static/js/html10n.js [hunk 1]
```diff
@@ -181,16 +181,20 @@ window.html10n = (function(window, document, undefined) {
...
+          for(l in data) { // try similar (e.g zh-hans)
+            if(lang != l && l.indexOf(lang) === 0 && data[l]) {
+              lang = l
+              break;
+            }
+          }
+          if(lang != l) return cb(new Error(msg))
         }
       }
-      if(lang != l) return cb(new Error(msg))
```
Comment (muxator) [inline src/static/js/html10n.js:193]: This code is clever.

Could you add a description _in natural language_ of what this section is doing?
An example would be ok.

Imagine to have a nitwit in front of you.

Comment (muxator) [reply src/static/js/html10n.js:193]: Also, the `var` is a bit flinky in such deep scopes in js, but in this case we can live without `let`. Just saying.

**Comment by muxator** [comment]
I have pulled in the first commit (7ec3be640b1a53e5334ad91309a52b8c65fcb52f), on respecting browser's preferred language on first pad load).

I am waiting for the comments on the code before merging the second one (on language variants).

Thanks.

**Comment by muxator** [comment]
@iquidus: it would be good to merge your work on language variants for **1.8.3**.

We have still some blockers in our [milestone](https://github.com/ether/etherpad-lite/milestone/13), so there is some time. If you are too busy, this last modification can be moved to the next release without problems.

Thanks.

**Comment by muxator** [comment]
Thanks @iquidus. I will merge this ASAP.

**Comment by muxator** [comment]
Merged, thanks.

**Pull request merged**